### ISSUE TITLE
Wrong indent with multiline type parameter

### ISSUE THREAD

**Issue by vRallev** [pinterest/ktlint#965]
I assumed that it would be fixed by #951, but I tried the latest snapshot and still see some broken behavior. 

## Expected Behavior
This is how Android Studio formats the code. We use 2 space indent, but the result is the same for 4 space indent.
```kotlin
  private fun RenderTester<Boolean, OverviewCategoryRowState, OverviewCategoryRowOutput,
    OverviewCategoryRendering>.expectVisibilityWorker(
    expectedOutput: Boolean? = null
  ) = apply {
  }
```
## Observed Behavior
KtLint complains about the indent in line 3 and 4:
```
Source.kt:3:5: Unexpected indentation (expected 6, actual 4)
Source.kt:4:3: Unexpected indentation (expected 4, actual 2)
```

If I change the code to what KtLint suggests, then the error changes
```kotlin
  private fun RenderTester<Boolean, OverviewCategoryRowState, OverviewCategoryRowOutput,
    OverviewCategoryRendering>.expectVisibilityWorker(
      expectedOutput: Boolean? = null
    ) = apply {
  }
```
```
Source.kt:3:1: Unexpected indentation (6) (should be 4)
Source.kt:4:1: Unexpected indentation (4) (should be 2)
```
## Your Environment
* Version of ktlint used: Latest snapshot from master 0.40.0

**Comment by cy6erGn0m** [comment]
Experiencing the same issue
